Make compatible with invest release/3.10 branch

davemfish opened the pull request:
Mainly, this involves accommodating the new arg types in the new `ARGS_SPEC`. There's also additional tests in `flaskapp.test.js` to render a `SetupTab` component for each model's actual `ARGS_SPEC`. That should help catch compatibility issues if we modify a model's `ARGS_SPEC`.

I also changed the keys in `ui_config.js` to use a different form of the model's name. I changed from the longform title (e.g. "SDR: Sediment Retention Ratio") to the "internal name" that is used by the CLI to `run` or `getspec` (e.g. "sdr"). That means the same name can be used to get the `ARGS_SPEC` and the UI spec. And it avoids relying on the `model_name` property of `ARGS_SPEC` - so in theory we might get rid of that property and have one less place with hardcoded model names. This change also facilitated the new batch of tests mentioned above.

This PR should is built around invest changes in https://github.com/natcap/invest/pull/623.

- [x] https://github.com/natcap/invest/pull/623. is merged

emlys commented on src/renderer/components/SetupTab/index.jsx:
```diff
@@ -50,9 +50,12 @@ function initializeArgValues(argsSpec, uiSpec, argsDict) {
     if (argsSpec[argkey].type === 'boolean') {
       value = argsDict[argkey] || false;
     } else if (argsSpec[argkey].type === 'option_string') {
+      const optionsArray = Array.isArray(argsSpec[argkey].options)
```
nice!

emlys commented on src/renderer/sampledata_registry.json:
```diff
@@ -1,133 +1,133 @@
   "Coastal Blue Carbon": {
     "filename": "CoastalBlueCarbon.zip",
-    "url": "https://www.googleapis.com/download/storage/v1/b/releases.naturalcapitalproject.org/o/invest%2F3.9.0.post432%2Bgdecbeffb%2Fdata%2FCoastalBlueCarbon.zip?generation=1622754766536120&alt=media",
+    "url": "https://www.googleapis.com/download/storage/v1/b/releases.naturalcapitalproject.org/o/invest%2F3.9.0.post721%2Bg0f8cc295%2Fdata%2FCoastalBlueCarbon.zip?generation=1629913657805202&alt=media",
```
I wish there was a way to avoid repeating these in every url, but that's not a problem for this PR

emlys commented on src/renderer/components/SetupTab/ArgInput/index.jsx:
```diff
@@ -125,15 +125,26 @@ export default class ArgInput extends React.PureComponent {
+          placeholderText = 'percent: a number from 0 - 100';
+          break;
+        case 'ratio':
+          placeholderText = 'ratio: a decimal from 0 - 1';
```
If it were me I'd just put `ratio (0 - 1)` and `percent (0 - 100)`, but I don't object to this.

davemfish replied:
Either way is probably good. I wanted to deter people from entering fractions, but that's pretty unlikely.

emlys reviewed (approved):
This all looks good to me! Switching away from the model display names is great, and even better if we can eventually get rid of one of the name types entirely.

davemfish commented on src/renderer/components/SetupTab/ArgInput/index.jsx:
```diff
@@ -125,15 +125,26 @@ export default class ArgInput extends React.PureComponent {
+      let placeholderText;
+      switch (argSpec.type) {
+        case 'freestyle_string':
+          placeholderText = 'string';
```
Yes, that's better!

davemfish commented:
> This all looks good to me! Switching away from the model display names is great, and even better if we can eventually get rid of one of the name types entirely.

Thanks for looking! I changed that one placeholder text to "text". Back to you!

davemfish commented:
Converted to draft for an attempt to switch from npm to yarn so that GHA will actually run.

emlys commented:
@davemfish this looks all good, just that one test is failing. Also does `yarn` still need `package-lock.json` to be checked in? I see there's a new file `yarn.lock` that may fill the same purpose?

davemfish commented:
> @davemfish this looks all good, just that one test is failing. Also does `yarn` still need `package-lock.json` to be checked in? I see there's a new file `yarn.lock` that may fill the same purpose?

Okay @emlys I did some debugging in the puppeteer test and it's passing. And I switched to `yarn` and removed the `package-lock.json`

I had some intermittent issues with other tests in `app.test.js`, but they're hard to reproduce. All are passing now, The next PR  (#167 ) ends up simplifying the most complicated tests in `app.test.js`, so I'm hopeful tests will be more reliable again there. And if not, I'll debug them there with the changes already in that PR.

emlys reviewed (approved):
Thanks @davemfish! Interested to see how it goes having `yarn` .
